runtime/src/main/java/io/quarkiverse/quinoa/QuinoaDevProxyHandler.java

```diff
@@ -58,8 +67,73 @@ public void handle(final RoutingContext ctx) {
             next(currentClassLoader, ctx);
             return;
         }
-        final String uri = computeURI(resourcePath, request);
+
         final MultiMap headers = request.headers();
+        final String uri = computeURI(resourcePath, request);
+
+        // WebSocket connection for Hot Module Reloading (HMR)
+        if (!config.prodMode && headers.get("Connection").contains("Upgrade")) {
```

Comment:
could we delegate to handling methods for classic request and websocket to make this method a bit shorter

Reply:
I refactored to delegate to two different methods.